src/sentry/models/appconnectbuilds.py

```diff
@@ -28,11 +28,11 @@ class AppConnectBuild(Model):
     project = FlexibleForeignKey("sentry.Project", db_constraint=False)
 
     # The integer ID of the app inside App Store Connect.
-    #
-    # TODO: On the AppStoreConnect API this is a str field, it was a mistake to make this
-    #    int.
     app_id = models.IntegerField()
 
+    # The integer ID of the app inside App Store Connect, as a string.
+    app_id_str = models.CharField(max_length=256)
```

Comment:
You'll need to include the default here. Currently it doesn't match the migration, so you're getting an error where it's detecting a difference between the model state and migration state.